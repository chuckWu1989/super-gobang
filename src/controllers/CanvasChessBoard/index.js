const drawLine = (ctx, s, e) => {
  ctx.beginPath();
  ctx.moveTo(s[0], s[1]);
  ctx.lineTo(e[0], e[1]);
  ctx.closePath();
  ctx.stroke();
};
export const drawBoard = (canvas, props) => {
  const {
    grid,
    coord,
    width,
    ratio,
  } = props;
  const ctx = canvas.getContext('2d');
  const height = width * ratio;
  ctx.clearRect(0, 0, width, height);
  for (let i = 0; i <= (grid + 1); i += 1) {
    drawLine(ctx, coord[0][i], coord[grid + 1][i]);
    drawLine(ctx, coord[i][0], coord[i][grid + 1]);
  }
};

export default drawBoard;
