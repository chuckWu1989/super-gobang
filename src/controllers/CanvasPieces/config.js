export const radiusRatio = 0.8;
export const s = 0 * Math.PI;
export const e = 2 * Math.PI;
