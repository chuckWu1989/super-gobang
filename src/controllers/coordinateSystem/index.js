export function setCoordinate() {
  const { width, ratio, grid } = this.props;
  if (typeof width !== 'number') {
    return false;
  }
  const coord = new Array(grid + 2);
  const trueWidth = Math.floor(width / 15) * 15;
  const trueHeight = Math.floor((trueWidth * ratio) / 15) * 15;
  const unitWidth = trueWidth / (grid + 1);
  const unitHeight = trueHeight / (grid + 1);
  for (let x = 0; x <= grid + 1; x += 1) {
    coord[x] = new Array(grid);
    for (let y = 0; y <= grid + 1; y += 1) {
      coord[x][y] = [unitWidth * x, unitHeight * y];
    }
  }
  return this.setState({ coord, width: trueWidth, height: trueHeight });
}

export default setCoordinate;
