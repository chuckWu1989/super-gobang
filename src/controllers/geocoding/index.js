/**
 * Calculate coordinates and set into state.
 */
export function setCoordinate() {
  const { width, ratio, grid } = this.props;
  if (typeof width !== 'number') {
    return false;
  }
  const coord = new Array(grid + 2);
  const unitWidth = Math.round(width / (grid + 1));
  const unitHeight = Math.round((width * ratio) / (grid + 1));
  const trueWidth = unitWidth * (grid + 1);
  const trueHeight = unitHeight * (grid + 1);
  for (let x = 0; x <= grid + 1; x += 1) {
    coord[x] = new Array(grid);
    for (let y = 0; y <= grid + 1; y += 1) {
      coord[x][y] = [unitWidth * x, unitHeight * y];
    }
  }
  return this.setState({
    coord,
    unitWidth,
    unitHeight,
    width: trueWidth,
    height: trueHeight,
  });
}
/**
 * Detect if the width of container is changed
 * @param {*} nextProps Next properties of the component
 * @param {*} props Current properties of the component
 */
export function detectWidthChanged(nextProps, props) {
  const { width: nextWidth } = nextProps;
  const { width } = props;
  if (nextWidth !== width) {
    setCoordinate.call(this);
  }
}

export default setCoordinate;
