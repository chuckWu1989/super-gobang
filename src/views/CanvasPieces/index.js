import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './style.less';

class CanvasPieces extends Component {
  static defaultProps = {
    detectChessMovesChange: () => {},
    detectWidthChange: () => {},
  }
  static propTypes = {
    width: PropTypes.number.isRequired,
    height: PropTypes.number.isRequired,
    detectChessMovesChange: PropTypes.func,
    detectWidthChange: PropTypes.func,
  }
  componentWillReceiveProps(nextProps) {
    this.props.detectChessMovesChange(this.canvas, nextProps);
  }
  componentDidUpdate(prevProps) {
    this.props.detectWidthChange(this.canvas, prevProps);
  }
  render() {
    const { width, height } = this.props;
    return (
      <canvas
        target
        ref={(self) => { this.canvas = self; }}
        width={width}
        height={height}
        className="canvas-pieces-style"
      />
    );
  }
}

export default CanvasPieces;
