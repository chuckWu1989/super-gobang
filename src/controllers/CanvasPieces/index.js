import { radiusRatio, s, e } from './config';

/**
 * Clear all content on chessboard
 * @param {*} ctx Context of the canvas
 * @param {number} width The width of the chessboard
 * @param {number} height The height of the chessboard
 */
export const clearCanvas = (ctx, width, height) => {
  ctx.clearRect(0, 0, width, height);
};
/**
 * Clear piece on chessboard
 * @param {*} ctx Context of the canvas
 * @param {array} coord The coordinate of piece
 * @param {number} unitWidth Unit width per grid
 */
export const clearPiece = (ctx, coord, unitWidth) => {
  const cx = coord[0];
  const cy = coord[1];
  ctx.clearRect(cx - (unitWidth / 2), cy - (unitWidth / 2), unitWidth, unitWidth);
};
/**
 * Draw piece on chessboard
 * @param {*} ctx Context of the canvas
 * @param {array} coord The coordinate of piece
 * @param {number} unitWidth Unit width per grid
 * @param {string} color The color of piece
 */
export const drawPiece = (ctx, coord, unitWidth, color) => {
  const r = radiusRatio * (unitWidth / 2);
  const cx = coord[0];
  const cy = coord[1];
  ctx.beginPath();
  ctx.fillStyle = color;
  ctx.arc(cx, cy, r, s, e);
  ctx.fill();
  ctx.closePath();
};
/**
 * Add or remove piece on chessboard
 * @param {*} ctx Context of the canvas
 * @param {object} nextProps Next properties of the component
 * @param {object} props Current properties of the component
 */
export const addOrRemovePiece = (ctx, nextProps, props) => {
  const { chessMoves: nextChessMoves, unitWidth, coord } = nextProps;
  const { chessMoves } = props;
  if (chessMoves.length < nextChessMoves.length) {
    const { x, y, color } = nextChessMoves[nextChessMoves.length - 1];
    const point = coord[x][y];
    drawPiece(ctx, point, unitWidth, color);
  } else if (chessMoves.length > nextChessMoves.length) {
    const pieces = chessMoves.slice(nextChessMoves.length, chessMoves.length);
    pieces.forEach((piece) => {
      const { x, y } = piece;
      const point = coord[x][y];
      clearPiece(ctx, point, unitWidth);
    });
  }
};
/**
 * Rerender pieces on chessboard
 * @param {*} ctx Context of the canvas
 * @param {*} props The properties of the component
 */
export const redrawPieces = (ctx, props) => {
  const {
    width,
    height,
    unitWidth,
    coord,
    chessMoves,
  } = props;
  clearCanvas(ctx, width, height);
  chessMoves.forEach((element) => {
    const { x, y, color } = element;
    const point = coord[x][y];
    drawPiece(ctx, point, unitWidth, color);
  });
};
/**
 * Detect if the chess moves
 * @param {element} container The element of the canvas
 * @param {object} nextProps Next properties of the component
 * @param {*} props Current properties of the component
 */
export const detectChessMovesChange = (container, nextProps, props) => {
  const ctx = container.getContext('2d');
  const { chessMoves } = props;
  const { chessMoves: nextChessMoves } = nextProps;
  if (chessMoves.length !== nextChessMoves.length) {
    addOrRemovePiece(ctx, nextProps, props);
  }
};
/**
 * Detect if the width of container is changed
 * @param {*} container The element of the canvas
 * @param {*} prevProps The previous properties of the component
 * @param {*} props Current properties of the component.
 */
export const detectWidthChange = (container, prevProps, props) => {
  const ctx = container.getContext('2d');
  const { width } = props;
  const { width: prevWidth } = prevProps;
  if (prevWidth !== width) {
    redrawPieces(ctx, prevProps);
  }
};

export default detectChessMovesChange;
