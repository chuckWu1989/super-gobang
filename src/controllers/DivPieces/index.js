import React from 'react';
import DivPiece from '../../views/DivPiece';
import { radiusRatio } from './config';

/**
 * Clear all content on chessboard
 * @param {array} children The children of component
 * @param {number} s Start point to be sliced
 * @param {number} e End point to be sliced
 * @returns {array} Updated children
 */
export function clearPiece(children, s, e) {
  const nextChildren = children.slice(s, e);
  return nextChildren;
}
/**
 * Draw piece on chessboard
 * @param {array} children The children of component
 * @param {array} coord The coordinate of piece
 * @param {number} unitWidth Unit width per grid
 * @param {string} color The color of piece
 * @returns {array} Updated children
 */
export const drawPiece = (children, coord, unitWidth, color) => {
  const size = radiusRatio * unitWidth;
  const x = (coord[0] - (size / 2)) + 1;
  const y = (coord[1] - (size / 2)) + 1;
  const key = `${x}-${y}-piece`;
  const piece = <DivPiece key={key} x={x} y={y} color={color} size={size} />;
  children.push(piece);
  return children;
};
/**
 * Rerender pieces on chessboard
 * @param {*} props The properties of the component
 */
export function redrawPieces(props) {
  const { unitWidth, coord, chessMoves } = props;
  const children = [];
  chessMoves.forEach((element) => {
    const { x, y, color } = element;
    const point = coord[x][y];
    drawPiece(children, point, unitWidth, color);
  });
  this.setState({ children });
}
/**
 * Add or remove piece on chessboard
 * @param {*} nextProps Next properties of the component
 * @param {*} props Current properties of the component
 */
export function addOrRemovePiece(nextProps, props) {
  const { chessMoves: nextChessMoves, unitWidth, coord } = nextProps;
  const { chessMoves } = props;
  const { children } = this.state;
  let nextChildren;
  if (chessMoves.length < nextChessMoves.length) {
    const { x, y, color } = nextChessMoves[nextChessMoves.length - 1];
    const point = coord[x][y];
    nextChildren = drawPiece(children, point, unitWidth, color);
  } else if (chessMoves.length > nextChessMoves.length) {
    const e = nextChessMoves.length - chessMoves.length;
    nextChildren = clearPiece(children, 0, e);
  }
  this.setState({ children: nextChildren.slice() });
}
/**
 * Detect if the chess moves
 * @param {object} nextProps Next properties of the component
 * @param {object} props Current properties of the component
 */
export function detectChessMovesChange(nextProps, props) {
  const { chessMoves: nextChessMoves } = nextProps;
  const { chessMoves } = props;
  if (nextChessMoves.length !== chessMoves.length) {
    addOrRemovePiece.call(this, nextProps, props);
  }
}
/**
 * Detect if the width of container is changed
 * @param {object} prevProps The previous properties of the component
 * @param {object} props Current properties of the component
 */
export function detectWidthChange(prevProps, props) {
  const { width } = props;
  const { width: prevWidth } = prevProps;
  if (prevWidth !== width) {
    redrawPieces.call(this, props);
  }
}

export default detectChessMovesChange;
