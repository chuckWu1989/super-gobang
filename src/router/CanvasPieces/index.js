import { detectChessMovesChange, detectWidthChange } from '../../controllers/CanvasPieces';
import CanvasPieces from '../../views/CanvasPieces';
import { connect } from '../../hocs/connect';

export const method = props => ({
  detectChessMovesChange: (canvas, nextProps) => detectChessMovesChange(canvas, nextProps, props),
  detectWidthChange: (canvas, prevProps) => detectWidthChange(canvas, props, prevProps),
});

export default connect(method)(CanvasPieces);
