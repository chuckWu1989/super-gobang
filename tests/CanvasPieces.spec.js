import React from 'react';
import toJson from 'enzyme-to-json';
import CanvasPieces from '../src/views/CanvasPieces';
import { method } from '../src/router/CanvasPieces';
import * as controller from '../src/controllers/CanvasPieces';

const { shallow } = global;
const unitWidth = 10;
const coord = Array.from({ length: 17 }, (v, i) => {
  const arr = [];
  for (let j = 0; j <= 16; j += 1) arr.push([i * 10, j * 10]);
  return arr;
});
const props = {
  width: 160,
  height: 160,
  unitWidth: 10,
  coord,
};

describe('CanvasChessPieces', () => {
  describe('view', () => {
    it('should be rendered properly', () => {
      const ownProps = { ...props };
      const tree = toJson(shallow(<CanvasPieces {...ownProps} />));
      expect(tree).toMatchSnapshot();
    });
    it('should call methods during lifecycle', () => {
      const ownProps = {
        ...props,
        detectChessMovesChange: jest.fn(),
        detectWidthChange: jest.fn(),
      };
      const wrapper = shallow(
        <CanvasPieces {...ownProps} />,
        { lifecycleExperimental: true },
      );
      wrapper.setProps({ width: 500 });
      expect(ownProps.detectChessMovesChange).toHaveBeenCalledTimes(1);
      expect(ownProps.detectWidthChange).toHaveBeenCalledTimes(1);
    });
  });
  describe('router', () => {
    it('should pass detectChessMovesChange', () => {
      expect('detectChessMovesChange' in method()).toBeTruthy();
    });
    it('should pass detectWidthChange', () => {
      expect('detectWidthChange' in method()).toBeTruthy();
    });
  });
  describe('controller', () => {
    let canvas;
    beforeEach(() => {
      canvas = document.createElement('canvas');
      canvas.width = props.width;
      canvas.height = props.height;
    });
    describe('clearCanvas', () => {
      it('should clear canvas in region of context', () => {
        const ctx = canvas.getContext('2d');
        controller.clearCanvas(ctx, props.width, props.height);
        const base64String = canvas.toDataURL();
        expect(base64String).toMatchSnapshot();
      });
    });
    describe('clearPiece', () => {
      it('should clear canvas in region of piece', () => {
        const coordinate = [50, 50];
        const ctx = canvas.getContext('2d');
        controller.clearPiece(ctx, coordinate, unitWidth);
        const base64String = canvas.toDataURL();
        expect(base64String).toMatchSnapshot();
      });
    });
    describe('drawPiece', () => {
      it('should draw a piece on the context', () => {
        const ctx = canvas.getContext('2d');
        const coordinate = [50, 50];
        const color = '#000000';
        controller.drawPiece(ctx, coordinate, unitWidth, color);
        const base64String = canvas.toDataURL();
        expect(base64String).toMatchSnapshot();
      });
    });
    describe('addOrRemovePiece', () => {
      it('should draw piece on context if the length of next chess moves is greater than current chess moves', () => {
        const ctx = canvas.getContext('2d');
        const nextProps = { chessMoves: [{ x: 5, y: 5, color: '#000000' }], unitWidth, coord };
        const currentProps = { chessMoves: [] };
        controller.addOrRemovePiece(ctx, nextProps, currentProps);
        const base64String = canvas.toDataURL();
        expect(base64String).toMatchSnapshot();
      });
      it('should remove pieces on context if the length of next chess moves is lower than current chess moves', () => {
        const ctx = canvas.getContext('2d');
        const coordinate = [50, 50];
        const color = '#000000';
        controller.drawPiece(ctx, coordinate, unitWidth, color);
        const nextProps = { chessMoves: [], unitWidth, coord };
        const currentProps = { chessMoves: [{ x: 5, y: 5, color: '#000000' }] };
        controller.addOrRemovePiece(ctx, nextProps, currentProps);
        const base64String = canvas.toDataURL();
        expect(base64String).toMatchSnapshot();
      });
      it('should skip if the length of next chess moves equals to current chess moves', () => {
        const ctx = canvas.getContext('2d');
        const nextProps = { chessMoves: [], unitWidth, coord };
        const currentProps = { chessMoves: [] };
        controller.addOrRemovePiece(ctx, nextProps, currentProps);
        const base64String = canvas.toDataURL();
        expect(base64String).toMatchSnapshot();
      });
    });
    describe('redrawPieces', () => {
      it('should redraw pieces on context', () => {
        const ownProps = {
          ...props,
          unitWidth,
          coord,
          chessMoves: [{ x: 5, y: 5, color: '#000000' }],
        };
        const ctx = canvas.getContext('2d');
        controller.redrawPieces(ctx, ownProps);
        const base64String = canvas.toDataURL();
        expect(base64String).toMatchSnapshot();
      });
    });
    describe('detectChessMovesChange', () => {
      it('should call addOrRemovePiece if the length of next chess moves is different with current chess moves', () => {
        const nextProps = { chessMoves: [{ x: 5, y: 5, color: '#000000' }], unitWidth, coord };
        const currentProps = { chessMoves: [] };
        controller.detectChessMovesChange(canvas, nextProps, currentProps);
        const base64String = canvas.toDataURL();
        expect(base64String).toMatchSnapshot();
      });
      it('should skip if the length of next chess moves equals to current chess moves', () => {
        const nextProps = { chessMoves: [], unitWidth, coord };
        const currentProps = { chessMoves: [] };
        controller.detectChessMovesChange(canvas, nextProps, currentProps);
        const base64String = canvas.toDataURL();
        expect(base64String).toMatchSnapshot();
      });
    });
    describe('detectWidthChange', () => {
      it('should call redraw if next width is different with current width', () => {
        const prevProps = {
          ...props,
          unitWidth,
          coord,
          chessMoves: [{ x: 5, y: 5, color: '#000000' }],
        };
        const currentProps = { width: 0 };
        controller.detectWidthChange(canvas, prevProps, currentProps);
        const base64String = canvas.toDataURL();
        expect(base64String).toMatchSnapshot();
      });
      it('should skip if next width is same with current width', () => {
        controller.detectWidthChange(canvas, props, props);
        const base64String = canvas.toDataURL();
        expect(base64String).toMatchSnapshot();
      });
    });
  });
});
