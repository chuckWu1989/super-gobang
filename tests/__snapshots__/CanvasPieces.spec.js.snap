// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CanvasChessPieces controller addOrRemovePiece should draw piece on context if the length of next chess moves is greater than current chess moves 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAKAAAACgCAYAAACLz2ctAAAABmJLR0QA/wD/AP+gvaeTAAAA1ElEQVR4nO3VMQrCUBAE0MHTCJaCtw6kTu9dxDaQlFq4NgYLxU+QvAfT7DRbLGwCAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADwckvRJrpW+ZtDcKcmc5PaSuTpoasjy+J4ZVtyLjRjz/gDHFffiA7u1F2Db/vkAz1928BPHJFOW73eqDprbJ+mSXCpdzQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAMgd+AokhW4FDDkAAAAASUVORK5CYII="`;

exports[`CanvasChessPieces controller addOrRemovePiece should remove pieces on context if the length of next chess moves is lower than current chess moves 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAKAAAACgCAYAAACLz2ctAAAABmJLR0QA/wD/AP+gvaeTAAAAeklEQVR4nO3BMQEAAADCoPVPbQ0PoAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAHgwkK8AAWW3G6sAAAAASUVORK5CYII="`;

exports[`CanvasChessPieces controller addOrRemovePiece should skip if the length of next chess moves equals to current chess moves 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAKAAAACgCAYAAACLz2ctAAAABmJLR0QA/wD/AP+gvaeTAAAAeklEQVR4nO3BMQEAAADCoPVPbQ0PoAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAHgwkK8AAWW3G6sAAAAASUVORK5CYII="`;

exports[`CanvasChessPieces controller clearCanvas should clear canvas in region of context 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAKAAAACgCAYAAACLz2ctAAAABmJLR0QA/wD/AP+gvaeTAAAAeklEQVR4nO3BMQEAAADCoPVPbQ0PoAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAHgwkK8AAWW3G6sAAAAASUVORK5CYII="`;

exports[`CanvasChessPieces controller clearPiece should clear canvas in region of piece 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAKAAAACgCAYAAACLz2ctAAAABmJLR0QA/wD/AP+gvaeTAAAAeklEQVR4nO3BMQEAAADCoPVPbQ0PoAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAHgwkK8AAWW3G6sAAAAASUVORK5CYII="`;

exports[`CanvasChessPieces controller detectChessMovesChange should call addOrRemovePiece if the length of next chess moves is different with current chess moves 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAKAAAACgCAYAAACLz2ctAAAABmJLR0QA/wD/AP+gvaeTAAAA1ElEQVR4nO3VMQrCUBAE0MHTCJaCtw6kTu9dxDaQlFq4NgYLxU+QvAfT7DRbLGwCAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADwckvRJrpW+ZtDcKcmc5PaSuTpoasjy+J4ZVtyLjRjz/gDHFffiA7u1F2Db/vkAz1928BPHJFOW73eqDprbJ+mSXCpdzQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAMgd+AokhW4FDDkAAAAASUVORK5CYII="`;

exports[`CanvasChessPieces controller detectChessMovesChange should skip if the length of next chess moves equals to current chess moves 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAKAAAACgCAYAAACLz2ctAAAABmJLR0QA/wD/AP+gvaeTAAAAeklEQVR4nO3BMQEAAADCoPVPbQ0PoAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAHgwkK8AAWW3G6sAAAAASUVORK5CYII="`;

exports[`CanvasChessPieces controller detectWidthChange should call redraw if next width is different with current width 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAKAAAACgCAYAAACLz2ctAAAABmJLR0QA/wD/AP+gvaeTAAAA1ElEQVR4nO3VMQrCUBAE0MHTCJaCtw6kTu9dxDaQlFq4NgYLxU+QvAfT7DRbLGwCAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADwckvRJrpW+ZtDcKcmc5PaSuTpoasjy+J4ZVtyLjRjz/gDHFffiA7u1F2Db/vkAz1928BPHJFOW73eqDprbJ+mSXCpdzQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAMgd+AokhW4FDDkAAAAASUVORK5CYII="`;

exports[`CanvasChessPieces controller drawPiece should draw a piece on the context 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAKAAAACgCAYAAACLz2ctAAAABmJLR0QA/wD/AP+gvaeTAAAA1ElEQVR4nO3VMQrCUBAE0MHTCJaCtw6kTu9dxDaQlFq4NgYLxU+QvAfT7DRbLGwCAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADwckvRJrpW+ZtDcKcmc5PaSuTpoasjy+J4ZVtyLjRjz/gDHFffiA7u1F2Db/vkAz1928BPHJFOW73eqDprbJ+mSXCpdzQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAMgd+AokhW4FDDkAAAAASUVORK5CYII="`;

exports[`CanvasChessPieces controller redrawPieces should redraw pieces on context 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAKAAAACgCAYAAACLz2ctAAAABmJLR0QA/wD/AP+gvaeTAAAA1ElEQVR4nO3VMQrCUBAE0MHTCJaCtw6kTu9dxDaQlFq4NgYLxU+QvAfT7DRbLGwCAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADwckvRJrpW+ZtDcKcmc5PaSuTpoasjy+J4ZVtyLjRjz/gDHFffiA7u1F2Db/vkAz1928BPHJFOW73eqDprbJ+mSXCpdzQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAMgd+AokhW4FDDkAAAAASUVORK5CYII="`;

exports[`CanvasChessPieces view should be rendered properly 1`] = `
<canvas
  className="canvas-pieces-style"
  height={160}
  target={true}
  width={160}
/>
`;
